### PR TITLE
carlos comments for chapter 6

### PR DESCRIPTION
**Pull request by TheRumle**
There are many big rewrites that we have not done as it would need to be moved somewhere else

### CODE REVIEW

File: report/sections/Frontend/client-server-interaction.tex [hunk 1]
```diff
@@ -14,17 +14,19 @@ \section{Client-Server Interaction}
 
 In the interest of reducing the number of database queries, the result of this request is cached on the client.
 The client will eagerly request new data to stay up to date with instructions or test code updates.
-We deemed that this short cache storage time is useful for when lecturers update test code during an exercise session.
+This short cache storage time is useful for when lecturers update test code during an exercise session.
 This way, all students have the latest test code shortly after the update while still minimizing unnecessary queries.
 
-When a user submits their exercise solution attempt, a mutation is sent to the Next.js backend.
-In this request, the necessary code and test code is included.
+When a user submits an exercise solution, a HTTP POST request is sent to the Next.js backend.
```
Comment (chhoumann) [inline report/sections/Frontend/client-server-interaction.tex:20]: I really disagree with changing mutation to HTTP POST request. We're using tRPC, so we use the terminology it uses.
Carlos wrote `what do you mean?` in his feedback, which tells me that he didn't read the preliminaries, where mutations are explained.

File: report/sections/Frontend/client-server-interaction.tex [hunk 2]
```diff
@@ -14,17 +14,19 @@ \section{Client-Server Interaction}
...
+In this request, the code and test is included.
 
-To handle this mutation, the Next.js server requests the Test Runner to process the given solution and test code.
+To handle this HTTP POST request, the Next.js server requests the Test Runner to process the given solution and test code.
```
Comment (chhoumann) [inline report/sections/Frontend/client-server-interaction.tex:23]: I really disagree with changing mutation to HTTP POST request. We're using tRPC, so we use the terminology it uses.
Carlos wrote `what do you mean?` in his feedback, which tells me that he didn't read the preliminaries, where mutations are explained.

**Review by chhoumann — CHANGES_REQUESTED** [review]

File: report/sections/Frontend/client-server-interaction.tex (hunk 1 above)
Comment (TheRumle) [reply report/sections/Frontend/client-server-interaction.tex:20]: maybe a compromise: a mutation is sent to the backend component in the form of an HTTP POST request.

Comment (chhoumann) [reply report/sections/Frontend/client-server-interaction.tex:20]: No, that confuses the terminology

Comment (chhoumann) [reply report/sections/Frontend/client-server-interaction.tex:20]:
```suggestion
When a user submits their exercise solution attempt, a mutation is sent to the Next.js backend.
```

File: report/sections/Frontend/client-server-interaction.tex (hunk 2 above)
Comment (chhoumann) [reply report/sections/Frontend/client-server-interaction.tex:23]:
```suggestion
To handle this mutation, the Next.js server requests the Test Runner to process the given solution and test code.
```

**Review by chhoumann — APPROVED** [review]

**Pull request merged**